crates/common/mqtt_channel/src/connection.rs

```diff
@@ -235,4 +253,22 @@ impl Connection {
     pub(crate) async fn do_pause() {
         sleep(Duration::from_secs(1)).await;
     }
+
+    pub(crate) async fn subscribe_to_topics(
+        mqtt_client: &AsyncClient,
+        subscriptions: Vec<rumqttc::SubscribeFilter>,
+    ) -> bool {
```

Comment:
This function must return a `Result<(), MqttError>`  and not a `bool`. Furthermore, this function should not `eprintln` anything: logging errors is the responsibility of the caller.